### PR TITLE
Added base api client for abstract content source

### PR DESCRIPTION
**Pull request by flying-pi**

### CODE REVIEW

File: bridge_adaptivity/api/backends/base_api_client.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    Provide base class for the content source API.
+    """
+
+    def __init__(self, content_source):
+        self.content_source = content_source
+        super().__init__(self.url)
+
+    @property
+    def url(self):
+        return self.content_source.host_url
```
Comment (idegtiarov) [inline bridge_adaptivity/api/backends/base_api_client.py]: lets have default param for api something like <host>/api/v1/

File: bridge_adaptivity/api/backends/base_api_client.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    def get_course_blocks(self, course_id):
```
Comment (idegtiarov) [inline bridge_adaptivity/api/backends/base_api_client.py]: This and following methods should be covered with the tests

File: bridge_adaptivity/api/backends/base_api_client.py (hunk 1 above)
Comment (flying-pi) [reply bridge_adaptivity/api/backends/base_api_client.py]: done

File: bridge_adaptivity/module/tests/test_views.py [hunk 3]
```diff
@@ -532,14 +532,21 @@ def setUp(self, mock_apply_async):
...
            return_value=[{'name': 'name'} for _ in range(10)])
-    def test_list_courses_multiple_sources(self, mock_get_provider_courses, mock_get_oauth_access_token):
+    @patch('api.backends.base_api_client.BaseApiClient.get_provider_courses',
+           return_value=[{'name': 'name'} for _ in range(10)])
+    def test_list_courses_multiple_sources(
+            self,
+            mock_base_get_provider_courses,
+            mock_get_edx_provider_courses,
+            mock_get_edx_oauth_access_token
+    ):
```
Comment (idegtiarov) [inline bridge_adaptivity/module/tests/test_views.py]: test description

File: .travis.yml [hunk 4]
```diff
@@ -17,7 +17,7 @@ before_script:
   - psql -c "CREATE USER bridge PASSWORD 'bridge-adaptivity';" -U postgres
   - psql -c "ALTER USER bridge CREATEDB;" -U postgres
   - psql -c "GRANT ALL PRIVILEGES on DATABASE traviscidb to bridge;" -U postgres
-
+  - find . | grep -E "(__pycache__|\.pyc|\.pyo$)" | xargs rm -rf
```
Comment (idegtiarov) [inline .travis.yml]: why do we need this string?

File: bridge_adaptivity/api/tests/test_views.py [hunk 5]
```diff
@@ -62,9 +62,16 @@ def test_base_api_client_calls(
         mock_base_get_provider_courses.assert_called_once()
         mock_edx_get_provider_courses.assert_not_called()
 
+    @patch('api.backends.edx_api_client.OpenEdxApiClient.get_oauth_access_token',
```
Comment (idegtiarov) [inline bridge_adaptivity/api/tests/test_views.py]: please move params to the new line if they do not fit one line

**Pull request merged**